app/sequencescape/sequencescape/api/v2/plate.rb

```diff
@@ -142,11 +139,31 @@ def assign_pools_to_wells
     wells.each do |well|
       pool = pooled_wells.find { |wells| wells.include?(well.location) }
       next if pool.nil?
-
       well.pool = pool
     end
   end
 
+  def purpose_config
+    Settings.purposes[purpose&.uuid] || {}
+  end
+
+  # return true if the plate has register_stock_plate flag in config file
+  def register_stock_plate?
+    purpose_config.fetch(:register_stock_plate, false)
+  end
+
+  # This method is used to register the stock plate in Sequencescape.
+  def register_stock
+    url = "#{self.class.site}/plates/#{id}/register_stock_for_plate"
+    response = Faraday.post(url)
```

Comment:
What's Faraday?

Normally we call v2 api endpoints like this:
Sequencescape::Api::V2::Plate.register_stock_for_plate

So this needs to be changed to match that standard.